src/client/search/AddSearchTerm.js

```diff
@@ -27,26 +27,32 @@ let AddSearchTerm = ({ dispatch }) => {
           $('#addSearchTerm').popup('hide');
           $('#addSearchTerm').popup('destroy');
           if (e.keyCode === 13) {
-            if (e.target.value.length >= 2) {
-              dispatch(addSearchTerm(e.target.value));
-              e.target.value = '';
-            } else {
+            if (e.target.value.length < 2) {
               $('#addSearchTerm').popup({
                 popup: '.negative.message.popup',
               });
+              $('#addSearchTermErrorText').html('Can only be at least 2 characters');
               $('#addSearchTerm').popup('show');
+            } else if (e.target.value.length >= 30) {
+              $('#addSearchTerm').popup({
+                popup: '.negative.message.popup',
+              });
+              $('#addSearchTermErrorText').html('Can only be less than 30 characters');
+              $('#addSearchTerm').popup('show');
+            } else {
+              dispatch(addSearchTerm(e.target.value));
+              e.target.value = '';
             }
           }
         }}
-
       />
       <div className="ui negative message popup transition hidden">
         <div className="header">
           <i className="red warning circle icon"></i>
           Search Validation Error
         </div>
-        <p>
-          Needs to be at least 2 characters
+        <p id="addSearchTermErrorText">
+          this text will be changed by line 37 and 43
```

Comment:
I know it *will* be changed but generally a failsafe message, even simply saying "An error has occurred", would be better imo